lint_test/avoid_unused_parameters_test.dart

```diff
@@ -10,11 +10,44 @@ import 'package:flutter/material.dart';
 
 /// Check the `avoid_unused_parameters` rule
 
+typedef MaxFun = int Function(int a, int b);
+
+// expect_lint: avoid_unused_parameters
+final MaxFun bad = (int a, int b) => 1;
+
+final MaxFun good = (int a, _) => a;
+
+final MaxFun ok = (int _, int __) => 1;
+
+final MaxFun goodMax = (int a, int b) {
+  return a + b;
+};
```

Comment:
Let's double-check how it works with named params -- we don't have the liberty to rename those to underscores.